packages/fast-usdc/src/exos/advancer.js

```diff
@@ -9,116 +11,219 @@ import { addressTools } from '../utils/address.js';
 
 /**
  * @import {HostInterface} from '@agoric/async-flow';
+ * @import {NatAmount} from '@agoric/ertp';
  * @import {ChainAddress, ChainHub, Denom, DenomAmount, OrchestrationAccount} from '@agoric/orchestration';
  * @import {VowTools} from '@agoric/vow';
  * @import {Zone} from '@agoric/zone';
  * @import {CctpTxEvidence, LogFn} from '../types.js';
  * @import {StatusManager} from './status-manager.js';
- * @import {TransactionFeedKit} from './transaction-feed.js';
  */
 
+/**
+ * Expected interface from LiquidityPool
+ *
+ * @typedef {{
+ *   lookupBalance(): NatAmount;
+ *   borrowUnderlying(amount: Amount<"nat">): Promise<PaymentPKeywordRecord>;
+ *   returnUnderlying(principalPayment: Payment<"nat">, feePayment: Payment<"nat">): Promise<void>
+ * }} AssetManagerFacet
+ */
+
+/**
+ * @typedef {{
+ *  chainHub: ChainHub;
+ *  log: LogFn;
+ *  statusManager: StatusManager;
+ *  vowTools: VowTools;
+ * }} AdvancerKitCaps
+ */
+
+/** type guards internal to the AdvancerKit */
+const WatcherHandlersShape = {
+  depositHandler: M.interface('DepositHandlerI', {
+    onFulfilled: M.call(AmountShape, ChainAddressShape).returns(VowShape),
+  }),
+  transferHandler: M.interface('TransferHandlerI', {
+    // TODO confirm undefined, and not bigint (sequence)
+    onFulfilled: M.call(M.undefined(), {
+      amount: AmountShape,
+      destination: ChainAddressShape,
+    }).returns(M.undefined()),
+    onRejected: M.call(M.error(), {
+      amount: AmountShape,
+      destination: ChainAddressShape,
+    }).returns(M.undefined()),
+  }),
+};
+
 /**
  * @param {Zone} zone
- * @param {object} caps
- * @param {ChainHub} caps.chainHub
- * @param {LogFn} caps.log
- * @param {StatusManager} caps.statusManager
- * @param {VowTools} caps.vowTools
+ * @param {AdvancerKitCaps} caps
```

Comment:
this name makes sense. Elsewhere we've called it `io`. I don't think the style guide takes a position

Reply:
`powers` is used similarly. Sometimes `tools`, though those are usually more closely related.

`caps` is not a dictionary word (or rather: it is, but it means "things you put on your head"). For very local usage, that doesn't matter much, but `AdvancerKitCaps` looks like something that might have a wider scope. Maybe `AdvancerKitPowers` instead?

Reply:
oh yeah, Powers. I think `powers` is strictly better than `caps`